### PR TITLE
week1

### PR DESCRIPTION
**Pull request by ArrogantGao**

### CODE REVIEW

**Comment by ArrogantGao** [comment]
Xuanzhao Gao

File: courseworks/week2/Xuanzhao_Gao/HW2.md [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+|    :----:   |    :----:   |    :----:   |    :----:   |    :----:   | :----:   |
+|ComplexF64| true | false | false | true | false | 
+|Complex{AbstractFloat}| true | false | false | false | false | 
+|Complex{<:AbstractFloat}| false | false | false | false | false | 
+|AbstractFloat| false | false | true | false | false | 
+|Union{Float64, ComplexF64}| false | false | false | false | false | 
+|Int32| true | true | false | true | false | 
+|Matrix{Float32}| true | false | false | false | true | 
+|Base.RefValue| false | false | false | false | true | 
+
```
[the remark this answers is not in the document]
Comment (GiggleLiu) [reply courseworks/week2/Xuanzhao_Gao/HW2.md:17]: ## About 3x+1.ipynb
1. Instead printing the result, the notebook is supposed to include a test.
2. The test should cover all Int32 integers.

File: courseworks/week1/XuanzhaoGao/HW1.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+2. Consider that spead of light given by $3 \times 10^8 ms^{-1}$, it takes 
+   $$
+   t_{signal} = 2 \times 10^{-1}/(3e8) \sim 6 \times 10^{-10} s = 0.6 ns
+   $$
+   and one loop on my CPU takes
+   $$
+   t_{loop} = 1/(2.6e9) \sim 3.8 \times 10^{-10} s
+   $$
+   and these two times are almost the same.
```
Comment (GiggleLiu) [inline courseworks/week1/XuanzhaoGao/HW1.md:14]: 👍🏾

File: courseworks/week1/XuanzhaoGao/lscpu.res [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+Vendor ID:                       GenuineIntel
+Model name:                      Intel(R) Xeon(R) Platinum 8378A CPU @ 3.00GHz
+CPU family:                      6
+Model:                           106
+Thread(s) per core:              2
+Core(s) per socket:              1
+Socket(s):                       1
+Stepping:                        6
+BogoMIPS:                        5999.99
+Flags:                           fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ss ht syscall nx pdpe1gb rdtscp lm constant_tsc rep_good nopl xtopology nonstop_tsc cpuid tsc_known_freq pni pclmulqdq ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand hypervisor lahf_lm abm 3dnowprefetch cpuid_fault invpcid_single ssbd ibrs ibpb stibp ibrs_enhanced fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves wbnoinvd arat avx512vbmi umip avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg avx512_vpopcntdq md_clear arch_capabilities
```
Comment (GiggleLiu) [inline courseworks/week1/XuanzhaoGao/lscpu.res:16]: You CPU has avx512 support. Its double precision floating point computing power is at least:
```
6 x 2.6 x 8 (avx512) x 2 (fma)
```
Measuring the CPU performance by benchmarking GEMM is more straight forward.

File: courseworks/week1/XuanzhaoGao/HW1.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+   NPU is a specialized circuit that implements all the necessary control and arithmetic logic necessary to execute machine learning algorithms. With its special designs, NPU do not need to transfer the data into the memory, but can directly put them into to its next layer, so that can greatly enhance the calculation speed and reduce the power cost.
+
+   FPGA is another kind of processor, which is an [integrated circuit](https://en.wikipedia.org/wiki/Integrated_circuit) designed to be configured by a customer or a designer after manufacturing. The most interesting preperity of FPGA is that the program on it can directly change the structure of its logic gate circuits, so that can run on a extremely high efficency.
+
+   TPU is designed by Google to speed up their tenserflow network, and it actually is a specially designed processor used to calculate matrix multiplication in a lower accuracy, which is of great need in AI training.
+
+4. Floating point number is not a Group
+
+   1. **Associativity of addition**: Yes
```
Comment (GiggleLiu) [inline courseworks/week1/XuanzhaoGao/HW1.md:26]: This answer should be fixed.

**Review by GiggleLiu — CHANGES_REQUESTED** [review]
Good job.

**Comment by ArrogantGao** [comment]
I remove the submission of week1 and week2 from my main branch and build a new branch

**Pull request closed without merging**